### PR TITLE
[Pimcore7] Remove php engine completely

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
- Remove any reference to templating
 - Switch to new Template Filename structure (@BundleName/Path/to.html.twig)
 - disable templating (twig enabled by default)
 - enable assets (was enabled by templating before)
 - remove all CompilerPasses related to PHP Engine
 - Switch references from Engine to Twig

### CODE REVIEW

**Review by dvesh3 — CHANGES_REQUESTED** [review]
Drop this as well https://github.com/pimcore/pimcore/blob/a4dcbe60df99109cf97c2cd6e5354639de606c1e/lib/Controller/Configuration/TemplatePhp.php#L23

- [ ] Doc updates missing

File: bundles/AdminBundle/Controller/Admin/EmailController.php [hunk 1]
```diff
@@ -129,9 +129,9 @@ public function showEmailLogAction(Request $request, ?Profiler $profiler)
         $emailLog = Tool\Email\Log::getById($request->get('id'));
 
         if ($request->get('type') == 'text') {
-            $templatingEnginePhp = $this->get('pimcore.templating.engine.php');
+            $templatingEngineTwig = $this->get('twig');
```
[the remark this answers is not in the document]
Comment (dpfaffenbauer) [reply bundles/AdminBundle/Controller/Admin/EmailController.php:132]: done

File: bundles/CoreBundle/Resources/config/pimcore/default.yml [hunk 2]
```diff
@@ -24,6 +22,7 @@ framework:
         handler_id:  null
     php_errors:
         log: true
+    assets: ~
 
 # Twig Configuration
 twig:
```
[the remark this answers is not in the document]
Comment (dpfaffenbauer) [reply bundles/CoreBundle/Resources/config/pimcore/default.yml:28]: done

**Comment by dpfaffenbauer** [comment]
@dvesh3 done

**Comment by brusch** [comment]
@dpfaffenbauer unfortunately travis is complaining, can you please check & fix? Thanks a lot! 😊

File: bundles/CoreBundle/Resources/config/pimcore/default.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvesh3) [reply bundles/CoreBundle/Resources/config/pimcore/default.yml:28]: @dpfaffenbauer tests are failing as path `/app/Resources/views` does not exist. I guess we need to move this config to demo & skeleton Repo. Sorry about the confusion! can you please revert it?

**Review by dvesh3 — CHANGES_REQUESTED** [review]
please change template paths for 
Ecommerce Trackers:
https://github.com/pimcore/pimcore/blob/288477cf6450b9317a75389422ae2975c5e81ef8/bundles/EcommerceFrameworkBundle/Tracking/Tracker.php#L89

https://github.com/pimcore/pimcore/blob/288477cf6450b9317a75389422ae2975c5e81ef8/bundles/EcommerceFrameworkBundle/Tracking/Tracker/Piwik.php#L84

https://github.com/pimcore/pimcore/blob/2662d44a8828ad8fc7c8d878020b115f8ba61d69/bundles/EcommerceFrameworkBundle/Tracking/Tracker/GoogleTagManager.php#L63

Others:
https://github.com/pimcore/pimcore/blob/3f966db4f8b5d4764e43b248cd131ede68128eb5/bundles/EcommerceFrameworkBundle/VoucherService/TokenManager/Pattern.php#L53

https://github.com/pimcore/pimcore/blob/3f966db4f8b5d4764e43b248cd131ede68128eb5/bundles/EcommerceFrameworkBundle/VoucherService/TokenManager/Single.php#L37

**Comment by dpfaffenbauer** [comment]
@dvesh3 tests are fixed, paths are also changed

File: bundles/AdminBundle/Resources/views/Admin/Asset/showVersionAudio.html.twig [hunk 3]
```diff
@@ -1 +1 @@
-{% include 'PimcoreAdminBundle:Admin/Asset:showVersionUnknown.html.twig' %}
\ No newline at end of file
+{% include '@PimcoreAdmin:Admin/Asset/showVersionUnknown.html.twig' %}
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/views/Admin/Asset/showVersionAudio.html.twig:1]:
```suggestion
{% include '@PimcoreAdmin/Admin/Asset/showVersionUnknown.html.twig' %}
```

File: bundles/AdminBundle/Resources/views/Admin/Asset/showVersionDocument.html.twig [hunk 4]
```diff
@@ -1 +1 @@
-{% include 'PimcoreAdminBundle:Admin/Asset:showVersionUnknown.html.twig' %}
\ No newline at end of file
+{% include '@PimcoreAdmin:Admin/Asset/showVersionUnknown.html.twig' %}
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/views/Admin/Asset/showVersionDocument.html.twig:1]:
```suggestion
{% include '@PimcoreAdmin/Admin/Asset/showVersionUnknown.html.twig' %}
```

File: bundles/AdminBundle/Resources/views/Admin/Asset/showVersionText.html.twig [hunk 5]
```diff
@@ -1 +1 @@
-{% include 'PimcoreAdminBundle:Admin/Asset:showVersionUnknown.html.twig' %}
\ No newline at end of file
+{% include '@PimcoreAdmin:Admin/Asset/showVersionUnknown.html.twig' %}
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/views/Admin/Asset/showVersionText.html.twig:1]:
```suggestion
{% include '@PimcoreAdmin/Admin/Asset/showVersionUnknown.html.twig' %}
```

File: bundles/AdminBundle/Resources/views/Admin/Asset/showVersionArchive.html.twig [hunk 6]
```diff
@@ -1 +1 @@
-{% include 'PimcoreAdminBundle:Admin/Asset:showVersionUnknown.html.twig' %}
\ No newline at end of file
+{% include '@PimcoreAdmin:Admin/Asset/showVersionUnknown.html.twig' %}
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/views/Admin/Asset/showVersionArchive.html.twig:1]:
```suggestion
{% include '@PimcoreAdmin/Admin/Asset/showVersionUnknown.html.twig' %}
```

File: bundles/AdminBundle/Resources/views/Admin/Asset/showVersionVideo.html.twig [hunk 7]
```diff
@@ -1 +1 @@
-{% include 'PimcoreAdminBundle:Admin/Asset:showVersionUnknown.html.twig' %}
\ No newline at end of file
+{% include '@PimcoreAdmin:Admin/Asset/showVersionUnknown.html.twig' %}
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/views/Admin/Asset/showVersionVideo.html.twig:1]:
```suggestion
{% include '@PimcoreAdmin/Admin/Asset/showVersionUnknown.html.twig' %}
```

File: bundles/AdminBundle/Controller/Admin/SettingsController.php [hunk 8]
```diff
@@ -1804,7 +1804,7 @@ public function testWeb2printAction(Request $request)
     {
         $this->checkPermission('web2print_settings');
 
-        $response = $this->render('PimcoreAdminBundle:Admin/Settings:testWeb2print.html.twig');
+        $response = $this->render('@PimcoreAdmin:Admin/Settings/testWeb2print.html.twig');
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/SettingsController.php:1807]:
```suggestion
        $response = $this->render('@PimcoreAdmin/Admin/Settings/testWeb2print.html.twig');
```

File: bundles/EcommerceFrameworkBundle/Tracking/Tracker/GoogleTagManager.php [hunk 9]
```diff
@@ -60,7 +60,7 @@ protected function configureOptions(OptionsResolver $resolver)
         parent::configureOptions($resolver);
 
         $resolver->setDefaults([
-            'template_prefix' => 'PimcoreEcommerceFrameworkBundle:Tracking/analytics/tagManager',
+            'template_prefix' => '@PimcoreEcommerceFramework/Tracking/analytics/tagManager',
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Tracking/Tracker/GoogleTagManager.php:63]: Change `'%s:%s.js.twig'` to `'%s/%s.js.twig'` getTemplatePath():
https://github.com/pimcore/pimcore/blob/288477cf6450b9317a75389422ae2975c5e81ef8/bundles/EcommerceFrameworkBundle/Tracking/Tracker.php#L89

**Comment by dpfaffenbauer** [comment]
all done

File: bundles/AdminBundle/Resources/views/Admin/Index/index.html.twig [hunk 10]
```diff
@@ -714,7 +716,7 @@
 {% endfor %}
 
 <!-- internal scripts -->
-{% if settings.disableMinifyJs %}
+{% if settings.disableMinifyJs or true %}
```
Comment (NiklasBr) [inline bundles/AdminBundle/Resources/views/Admin/Index/index.html.twig:719]: Should `or true` be here?

File: bundles/EcommerceFrameworkBundle/Tracking/Tracker.php [hunk 11]
```diff
@@ -86,7 +86,7 @@ protected function configureOptions(OptionsResolver $resolver)
     protected function getTemplatePath(string $name)
     {
         return sprintf(
-            '%s:%s.js.twig',
+            '%s:/%s.js.twig',
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Tracking/Tracker.php:89]:
```suggestion
            '%s/%s.js.twig',
```

File: bundles/AdminBundle/Resources/views/Admin/Index/index.html.twig (hunk 10 above)
Comment (dpfaffenbauer) [reply bundles/AdminBundle/Resources/views/Admin/Index/index.html.twig:719]: should be removed ;)

Comment (dpfaffenbauer) [reply bundles/AdminBundle/Resources/views/Admin/Index/index.html.twig]: is removed

**Comment by dpfaffenbauer** [comment]
@dvesh3 done again :)

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@dpfaffenbauer thanks a lot! 🙏

**Comment by dvesh3** [comment]
WIP: We still need a templating service instead of direct twig references in the core, basically to support PHP templating bundle.

**Comment by dpfaffenbauer** [comment]
@dvesh3 Well, you need some type of delegate. But as soon as you enable php engine, you'll get loads of deprecation messages, that is what I tried to reduce with this PR. So if Pimcore just has a delegation renderer, without using the ones from symfony, that is fine with me.

**Pull request merged**